scrapy/utils/trackref.py

```diff
@@ -9,27 +9,28 @@
 alias to object in that case).
 """
 
+from __future__ import annotations
+
 from collections import defaultdict
 from operator import itemgetter
 from time import time
-from typing import TYPE_CHECKING, Any, DefaultDict, Iterable
+from typing import TYPE_CHECKING, Any
 from weakref import WeakKeyDictionary
 
 if TYPE_CHECKING:
-    # typing.Self requires Python 3.11
-    from typing_extensions import Self
+    from collections.abc import Iterable
 
 
 NoneType = type(None)
-live_refs: DefaultDict[type, WeakKeyDictionary] = defaultdict(WeakKeyDictionary)
+live_refs: defaultdict[type, WeakKeyDictionary] = defaultdict(WeakKeyDictionary)
 
 
 class object_ref:
     """Inherit from this class to a keep a record of live instances"""
 
     __slots__ = ()
 
-    def __new__(cls, *args: Any, **kwargs: Any) -> "Self":
+    def __new__(cls, *args: Any, **kwargs: Any) -> object_ref:
```

Comment:
I don't this is correct, `object.__new__(cls)` returns a `cls` instance.

Reply:
Reverted. But with quotes removed. I got an error message about "Self" being a string literal.